vision_evaluation/evaluators.py

```diff
@@ -816,3 +817,182 @@ def __init__(self):
         super().__init__(metric='SPICE')
         self.predictions = []
         self.targets = []
+
+
+class MattingEvaluatorBase(Evaluator):
+    """
+    Base class for image matting evaluator
+    """
+    def __init__(self):
+        super(MattingEvaluatorBase, self).__init__()
+        self.predictions = []
+        self.targets = []
+        self.metric = None
+
+    def add_predictions(self, predictions, targets):
+        """ Evaluate list of image with image matting results
+        Args:
+            predictions: list of image matting predictions, [mating1, mating2, ...]
+            targets: list of image matting ground truth, [gt1, gt2, ...]
+        """
+        self.targets += targets
+        self.predictions += predictions
+
+    def reset(self):
+        super(MattingEvaluatorBase, self).reset()
+        self.targets = []
+        self.predictions = []
+
+    def _convert2binary(self, mask, threshold=128):
+        bin_mask = mask.copy()
+        bin_mask[mask < threshold] = 0
+        bin_mask[mask >= threshold] = 1
+        return bin_mask
+
+    def _find_contours(self, matting, thickness=10):
+        matting = np.copy(matting)
+        opencv_major_version = int(cv2.__version__.split('.')[0])
+        if opencv_major_version >= 4:
+            contours, _ = cv2.findContours(matting, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+        else:
+            _, contours, _ = cv2.findContours(matting, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+        mask = np.zeros(matting.shape, np.uint8)
+
+        cv2.drawContours(mask, contours, -1, 255, thickness)
+        return mask
+
+    def _create_contour_mask(self, gt_mask, pred_mask, line_width=10):
+        contour_mask = self._find_contours((gt_mask * 255).astype('uint8'), thickness=line_width) / 255.0
+        gt_contour_mask = gt_mask * contour_mask
+        pred_contour_mask = pred_mask * contour_mask
+        return gt_contour_mask, pred_contour_mask
+
+
+class MeanIOUEvaluator(MattingEvaluatorBase):
```

Comment:
seem all four evaluators can calculate intermediate results instead of remembering all ground truth and predictions